cmd/main.go

```diff
@@ -167,67 +160,75 @@ func loadConfigs() (*config.Config, error) {
 		}
 		mqttTopic := mainflux.Env(envMqttChannel, defMqttChannel)
 		natsTopic := "*"
-		rc := []config.Route{{
+		rc := []exp.Route{{
 			MqttTopic: mqttTopic,
 			NatsTopic: natsTopic,
 		}}
-		cfg := config.NewConfig(sc, rc, mc, configFile)
-		err = loadCertificate(cfg)
+
+		cfg := exp.Config{
+			Server: sc,
+			Routes: rc,
+			MQTT:   mc,
+			File:   configFile,
+		}
+		mqtt, err := loadCertificate(cfg.MQTT)
+		cfg.MQTT = mqtt
 		if err != nil {
 			return cfg, err
 		}
-		err = cfg.Save()
-		if err != nil {
+
+		if err := exp.Save(cfg); err != nil {
 			log.Println(fmt.Sprintf("Failed to save %s", err))
 		}
 		log.Println(fmt.Sprintf("Configuration loaded from environment, initial %s saved", configFile))
 		return cfg, nil
 	}
-	err := loadCertificate(cfg)
+	mqtt, err := loadCertificate(cfg.MQTT)
+	cfg.MQTT = mqtt
```

Comment:
Is it a good idea to set `cfg.MQTT` if an error happened or this line should be after error check?